BUG-ID:CLOUDSTACK-9331:added code in marvin frame&new config file for advBaremetal support …

cloudsadhu opened the pull request:
Added code  in marvin framework&new config file  to support baremetal advanced testcases

DaanHoogland commented:
tools/marvin/marvin/config/Baremtal_AdvZone_Vmware.cfg does not have a license. @cloudsadhu please add it.

shwetaag commented on tools/marvin/marvin/config/Baremtal_AdvZone_Vmware.cfg:
```diff
@@ -0,0 +1,191 @@
+                                    "cpunumber": "1",
+                                    "cpuspeed": "1000",
+                                    "memory": "1024",
+									"hosttags": "host15"
```
indentation issues

cloudsadhu replied:
Done, thanks.

shwetaag commented on tools/marvin/marvin/config/Baremtal_AdvZone_Vmware.cfg:
```diff
@@ -0,0 +1,191 @@
+                }
+            ],
+            "internaldns1": "10.x.x.x",
+			"baremetalrcturl": "http://10.x.x.x/baremetal/sadhurct1.json_old",
```
convert sadhurct1 filename to a more meaningful name

cloudsadhu replied:
Done,Thank you.

shwetaag commented on tools/marvin/marvin/config/Baremtal_AdvZone_Vmware.cfg:
```diff
@@ -0,0 +1,191 @@
+			"baremetalrcturl": "http://10.x.x.x/baremetal/sadhurct1.json_old",
+            "secondaryStorages": [
+                {
+                    "url": "nfs://10.x.x.x:/export/home/sadhu/secrc2",
```
its good if you can remove your name in path both in secondary and primary storage path

cloudsadhu replied:
Done,Thank you.

cristofolini commented:
Shouldn't the path shown in line 55 of your config file be OS-independent?
I'd also suggest fixing the typo in the file name. At least I believe it was meant to be Barem**e**tal_AdvZone_Vmware.cfg instead of `Baremtal_AdvZone_Vmware.cfg`.
If you decide to fix these things, it would be nice if you squashed your commits into a single one, since they're all implementing the same feature. :)

cloudsadhu commented:
I have incorporated the all the  review comments and squashed all my commits  to single commit.

This  script prepares  setup required  to run advanced bare metal  testcases ,i.e that creates advanced  zone setup with vmware& bare metal cluster. prepares the system vms& upload the switc configuration  file required for baremetal.

Test results:

C:\Python27\python.exe D:/cpmsource/acs/cloudstack/tools/marvin/marvin/deployDataCenter.py -i D:\cpmsource\internal-cloudstack\BVT_SmokeTest_Adv_Vmware.cfg

==== Log Folder Path: tmp//DeployDataCenter__Apr_05_2016_16_34_44_B2WTSH. All logs will be available here ====

==== Deploy DC Started ====

=== Data Center Settings are dumped to tmp//DeployDataCenter__Apr_05_2016_16_34_44_B2WTSH/dc_entries.obj===

====Deploy DC Successful=====

shwetaag commented:
LGTM  based on code walk through

swill commented:
I don't have the ability to test this.  Would you mind running the following tests to verify there are no issues in the setup once the DC has been deployed and then post back the results?

```
echo "Running tests with required_hardware=true"
nosetests --with-marvin --marvin-config=${marvinCfg} -s -a tags=advanced,required_hardware=true \
smoke/test_password_server.py \
smoke/test_vpc_redundant.py \
smoke/test_routers_iptables_default_policy.py \
smoke/test_routers_network_ops.py \
smoke/test_vpc_router_nics.py \
smoke/test_router_dhcphosts.py \
smoke/test_loadbalance.py \
smoke/test_internal_lb.py \
smoke/test_ssvm.py \
smoke/test_vpc_vpn.py \
smoke/test_privategw_acl.py \
smoke/test_network.py

echo "Running tests with required_hardware=false"
nosetests --with-marvin --marvin-config=${marvinCfg} -s -a tags=advanced,required_hardware=false \
smoke/test_routers.py \
smoke/test_network_acl.py \
smoke/test_reset_vm_on_reboot.py \
smoke/test_vm_life_cycle.py \
smoke/test_service_offerings.py \
smoke/test_network.py \
component/test_vpc_offerings.py \
component/test_vpc_routers.py
```

cloudsadhu commented:
HI,
Thanks for your review comment  and i  ran the   few  tests  on the created setup  even though it won't impact the any testcases but for your request i ran  few test(disk_offering  tests)

Its only marvin changes which will help to prepare  the setup and won't impact the test cases.

also noticed one issue in cfg file while running these cases which i fixed now with new commit.fc98a3fabe489d1f4b6addf4836c1b3c257fae1a(logger key  defined inside Zone dictionary but i should define outside so i fixed that part). please find the test results as well.
Test to create disk offering ... === TestName: test_01_create_disk_offering | Status : SUCCESS ===
ok
Test to create  a sparse type disk offering ... === TestName: test_02_create_sparse_type_disk_offering | Status : SUCCESS ===
ok
Test to create  a sparse type disk offering ... === TestName: test_04_create_fat_type_disk_offering | Status : SUCCESS ===
ok
Test to update existing disk offering ... === TestName: test_02_edit_disk_offering | Status : SUCCESS ===
ok
Test to delete disk offering ... === TestName: test_03_delete_disk_offering | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 5 tests in 842.283s

OK

sanju1010 commented:
LGTM based on the code review!!

swill commented:
I think this one is ready.  I want Jenkins to be green before I merge though.  I will review what is going on with Jenkins today.  I will this to my merge list.

cloudsadhu commented:
Thanks Swill  and  now All the checks are passed.Please merge this PR.
